### DISCUSSION TITLE
Unable to set speed (RPM) on MFM drives

### DISCUSSION THREAD

**Discussion by jester-xbmc** [86Box/86Box#4917] [Ideas]
### What happened?

While duplicating my real hardware as close as possible I wanted to add an MFM drive (trying to emulate a Miniscribe MFM drive on an Olivetti M240) I noticed that one can only select Ram disk as the speed, and not RPM, in my case 3600 RPM, like you can for IDE, manually adding the speed line into the cfg line doesn't work either, was this by design ?

It would be nice to be able to emulate the correct speed for MFM drives.

![image](https://github.com/user-attachments/assets/419aed33-63a8-4de2-b6f1-f798a751d0cc)

### Configuration file

```ini
not linked
```

### Operating system

WIndows 10

### CPU

Intel

### 86Box version

4.2.1 or latest from main

### Build architecture

Windows - x64 (64-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Download source

Official website (Jenkins, GitHub)

**Comment by OBattler** [comment]
NOTABUG, RPM selection is not yet implemented for those old drives.

**Reply by jester-xbmc** [reply]
any implementation timeline on this ?

now that we are adding all kinds of IDE Hard drives....

**Comment by jriwanek** [comment]
Small correction: "now that an external contributor added..."
If you'd like to assist, it'd be most welcome.

**Reply by jester-xbmc** [reply]
No need to get pedantic, the "we" is a type of expression, I'm not implying that I'm a 86box dev.
I was merely asking if there was a timeline, somebody might already be looking at this...

I currently have enough on my plate elsewhere but fair, I'll keep it in mind.
Thanks.

**Comment by jester-xbmc** [comment]
I've looked at the code (not that I have any clue) I'm happy to start added MFM drive definitions in https://github.com/86Box/86Box/blob/master/src/disk/hdd.c however I think the logic to insert/read speed settings is missing in https://github.com/86Box/86Box/blob/master/src/disk/hdc_st506_xt.c which I assume is used for XT MFM drives, correct ?

**Reply by OBattler** [reply]
Yes, correct.

**Reply by jester-xbmc** [reply]
I'm looking at this again, as now there is floppy/hard disk sound support it would be good to get MFM HD sounds in, however as the RPM isnt implemented yet MFM sounds probably won't work either ?

is this the right way going about doing this ? @OBattler 

https://github.com/86Box/86Box/compare/master...jester-xbmc:86Box:master

**Comment by jester-xbmc** [comment]
Ping @OBattler when you have a moment, could you please have a look to see if this is a way to go or utter crap? before I even do a PR ?

Thanks :)

https://github.com/86Box/86Box/compare/master...jester-xbmc:86Box:master

**Reply by OBattler** [reply]
Seems fine to me.

**Reply by jester-xbmc** [reply]
Thanks, appreciated, I'll push a PR

**Comment by jester-xbmc** [comment]
Merged in https://github.com/86Box/86Box/pull/6885

closing
